### PR TITLE
164 remove cyclic dep

### LINKED ISSUE #164: Remove Cyclic Dependencies

### PR DESCRIPTION
**Pull request by mpsijm**
Will close #164 

Okay guys, some moving around has been done again!

- The buttonSound player is moved from MainMenuScene to AbstractMenu/Popup
- The SoundEffectObserver is now an inner class of GameScene
- The CollisionMap is moved from `util` to `level`
- KeyMap has been refactored to remove dependency on GamePlay
- Move ShaftEscape.OBSERVABLE to GameObservable
- Move Highscores class from `level.state` to `server`
- Rename Highscores class to HighscoreAdapter
- Move Highscore class to own file

I recommend reviewing this PR per commit, because that'll give you much more overview. It's a bit more work, but it's not like much logic changed :)
Good luck reviewing!

<review, naming, organization

### CODE REVIEW

File: src/main/java/nl/tudelft/ti2206/group9/ShaftEscape.java [hunk 1]
```diff
@@ -32,17 +32,9 @@
     public static final int WIDTH = 480 - 16;
```
Comment (thervh70) [inline src/main/java/nl/tudelft/ti2206/group9/ShaftEscape.java]: *Auto-generated comment*

    '16' is a magic number.

**Comment by mdingjan** [comment]
I reviewed your changes and I agree with them. 
There are no SAT checkstyle violations left (except for the one @thervh70 described and two similar violations (PlayerRenderer and GameScene)).

Have one more look at them, add a coment for the 'answers' of the calculations which explain the answers. After that, ready for merging!

<review,logic

File: src/main/java/nl/tudelft/ti2206/group9/gui/scene/GameScene.java [hunk 2]
```diff
@@ -281,4 +304,75 @@ public void update(final Observable o, final Object arg) {
...
 
+    /**
+     * The SoundEffectsPlayer plays the SoundEffects that accompany the player's
+     * movement.
+     * @author Maarten and Mitchell
+     */
+    private static class SoundEffectObserver implements GameObserver {
+
+        /** Constant which is used for increasing the soundtrack speed. */
+        private static final double SPEED_INCREASE = Math.pow(2, 1. / 12.);
```
Comment (mdingjan) [inline src/main/java/nl/tudelft/ti2206/group9/gui/scene/GameScene.java]: This one, '12' is a magic number.

<review

**Comment by thervh70** [comment]
I agree with @mdingjan. This is looking good.

< review

**Comment by MathiasMeuleman** [comment]
I agree too. Please fix, then we can merge!

<review

**Pull request merged**